fido2-net-lib/AuthenticatorAttestationResponse.cs

```diff
@@ -306,12 +317,23 @@ public AttestationVerificationData Verify(CredentialCreateOptions options, strin
                     // If neither x5c nor ecdaaKeyId is present, self attestation is in use
                     else
                     {
-                        var packedCert = new X509Certificate2(attData.credentialPublicKey.ToArray());
-                        var packedPubKey = (ECDsaCng)packedCert.GetECDsaPublicKey();
                         // Validate that alg matches the algorithm of the credentialPublicKey in authenticatorData
-                        if (true != algMap[alg.AsInt32()].Equals(packedPubKey.Key.Algorithm)) throw new Fido2VerificationException();
+                        var COSE_kty = credentialPublicKey[PeterO.Cbor.CBORObject.FromObject(1)]; // 2 == EC2
+                        if (!credentialPublicKey[PeterO.Cbor.CBORObject.FromObject(3)].Equals(alg)) throw new Fido2VerificationException("Algorithm mismatch");
+                        var COSE_crv = credentialPublicKey[PeterO.Cbor.CBORObject.FromObject(-1)]; // 1 == P-256 curve 
+                        var x = credentialPublicKey[PeterO.Cbor.CBORObject.FromObject(-2)].GetByteString();
+                        var y = credentialPublicKey[PeterO.Cbor.CBORObject.FromObject(-3)].GetByteString();
+                        var cng = ECDsaCng.Create(new ECParameters
+                        {
+                            Curve = ECCurve.NamedCurves.nistP256,
```

Comment:
@aseigler Will this always be ECC nistP256?

Reply:
No...working through conformance tests.

Reply:
👍 Got it!